### ISSUE TITLE
[Feature] use chatgpt-next-web as a note taking tool

### ISSUE THREAD

**Issue by jkjoker** [Yidadaa/ChatGPT-Next-Web#3394]
I want to use this app as a note taking tool. Therefore, some of the functions inside can be designed like note taking tools.

For example:

When exporting notes, it is important to keep the timestamp as time is an important component of the notes.

Notes can be automatically synchronized.

The conversation list on the left. When clicking on synchronization, its order can always be local. Will not be covered or disrupted.

The list of note taking sessions on the left. Supports topping.

When we haven't used any models or called any APIs. Messages can be sent to a certain conversation, and these messages can also be retrieved. Can support export.

In short, we can use this app as a note taking tool, and we may not call any models or APIs inside. At this point, what we need is effective synchronization functionality and some basic editing support. 

I will add more thoughts here in the future. Thank you, thank you very much for having such a tool. It is this tool that has prompted me to think carefully.

**Comment by H0llyW00dzZ** [comment]
can you explain more in scenario about note taking tools (e.g, provide in a screenshot about how it work for a note taking tools) ?

**Comment by jkjoker** [comment]
Thank you for being interested in my ideas. I can provide an example, but before that, what I want to say is that I have been thinking about using ChatGPT next web as a note taking tool for some time, and now I am truly putting this idea into practice.

Previously, in ChatGPT next web, I learned new things through session windows one by one. I have found that a lot of content in a conversation is of high quality and needs to be saved. Interestingly, ChatGPT next web supports exporting these contents, and even now, it has added synchronization functionality. Once configured, the content can be automatically saved in the cloud without having to be exported to other note taking tools. So, for me, ChatGPT next web is already a note taking tool.

However. In many aspects of design details, I can feel that ChatGPT next web is still different from its note taking tool. After all, this tool did not originally exist as a note taking tool. This is also the reason why I propose suggestions here. As a heavy note taking user, I feel that ChatGPT next web. It is very likely to become the most needed note taking tool for learning. More specifically. This tool is convenient for us to call AI. This tool manages our notes in a conversational manner. Although we have many note taking apps that we can use, other apps cannot achieve the above two points.

However, an undeniable fact before us is. In the future, we will have more and more AI to call. Many of our content is destined to exist in the form of conversations.

In short. Although there may not be many people using ChatGPT next web as a note taking tool yet, ChatGPT next web has actually done a lot of things that note taking tools should do.

Let's imagine what kind of effect we would see if some small details could be optimized from the perspective of note taking tools.

Firstly, let's take a look at what would happen if we didn't call any models or even connect to the internet now, and only optimized some details.

1. The conversation box can be placed at the top.
2. The note content can be automatically synchronized.
3. Users can input through voice.
4. Users can input images.
5. Users can customize various preset replies.
6. When users export content, timestamp information can also be attached.

We will only mention these 6 points for now, if they can be realized. So, in fact, it has already been achieved that for the same topic, the content information is gathered together and arranged in timestamp order, and can be exported when needed. This has replaced what a note taking tool is doing.

We are still not connected to the internet and do not call any models. And just make a little more optimization based on the above, and see what happens.

The search function we need to optimize now is to enable us to search for notes. If this were to be achieved, our notes would not even need to be exported at all.

Alright, now let's consider the AI issue again. Now we will call AI. What happens now when calling AI based on one's own notes?

We will find a conversation can become very, very long. Each item in the conversation can be continuously edited and improved. Every piece of content can be modified, polished, and delved into through AI. Perhaps there is another possibility. That is, every conversation we have will become an exclusive learning assistant.

Well, alright, these are some of my current thoughts, and that's probably all. Obviously, if we continue to think, there will be many more ways to play. In short, I believe that everyone will have a large amount of session data to manage. The management of session data is a capability that current note taking tools do not possess. More importantly, as mentioned repeatedly earlier, ChatGPT next web is actually a decent note taking tool now.

Finally, I would like to attach my usage scenario:

![Snipaste_2023-11-27_20-30-30](https://github.com/Yidadaa/ChatGPT-Next-Web/assets/48642801/90a182cf-1b18-4a33-ac7d-5fb377dceeb8)
![Snipaste_2023-11-27_20-32-10](https://github.com/Yidadaa/ChatGPT-Next-Web/assets/48642801/7e8ade52-860b-470b-94c7-fb4c480d69c9)

As shown in the figure, I filled in the wrong API address in the settings to avoid calling AI. Then I input my feedback on ChatGPT from the dialog box. After receiving your reply, I added it. This completes a recording.

Oh, that's not right. I still used a note taking tool because my screenshots must be placed inside the note taking tool, After all, it's not possible to put pictures in ChatGPT next web yet (From this, it can be seen that when designing, if there is a note taking tool perspective, many functions may be designed differently, such as the image function. Who said that sending images must call AI? Can't we just save the images? Saving images is a note taking requirement, and the information on the images is useful to us, regardless of whether AI has the ability to interpret images.)

I think that if many details can be optimized and designed from the perspective of note taking tools, the final effect will definitely be very stunning. I am looking forward to it very much. Thank you!

**Comment by jkjoker** [comment]
> can you explain more in scenario about note taking tools (e.g, provide in a screenshot about how it work for a note taking tools as example) ?


Thank you for being interested in my ideas. I can provide an example, but before that, what I want to say is that I have been thinking about using ChatGPT next web as a note taking tool for some time, and now I am truly putting this idea into practice.

Previously, in ChatGPT next web, I learned new things through session windows one by one. I have found that a lot of content in a conversation is of high quality and needs to be saved. Interestingly, ChatGPT next web supports exporting these contents, and even now, it has added synchronization functionality. Once configured, the content can be automatically saved in the cloud without having to be exported to other note taking tools. So, for me, ChatGPT next web is already a note taking tool.

However. In many aspects of design details, I can feel that ChatGPT next web is still different from its note taking tool. After all, this tool did not originally exist as a note taking tool. This is also the reason why I propose suggestions here. As a heavy note taking user, I feel that ChatGPT next web. It is very likely to become the most needed note taking tool for learning. More specifically. This tool is convenient for us to call AI. This tool manages our notes in a conversational manner. Although we have many note taking apps that we can use, other apps cannot achieve the above two points.

However, an undeniable fact before us is. In the future, we will have more and more AI to call. Many of our content is destined to exist in the form of conversations.

In short. Although there may not be many people using ChatGPT next web as a note taking tool yet, ChatGPT next web has actually done a lot of things that note taking tools should do.

Let's imagine what kind of effect we would see if some small details could be optimized from the perspective of note taking tools.

Firstly, let's take a look at what would happen if we didn't call any models or even connect to the internet now, and only optimized some details.

1. The conversation box can be placed at the top.
2. The note content can be automatically synchronized.
3. Users can input through voice.
4. Users can input images.
5. Users can customize various preset replies.
6. When users export content, timestamp information can also be attached.

We will only mention these 6 points for now, if they can be realized. So, in fact, it has already been achieved that for the same topic, the content information is gathered together and arranged in timestamp order, and can be exported when needed. This has replaced what a note taking tool is doing.

We are still not connected to the internet and do not call any models. And just make a little more optimization based on the above, and see what happens.

The search function we need to optimize now is to enable us to search for notes. If this were to be achieved, our notes would not even need to be exported at all.

Alright, now let's consider the AI issue again. Now we will call AI. What happens now when calling AI based on one's own notes?

We will find a conversation can become very, very long. Each item in the conversation can be continuously edited and improved. Every piece of content can be modified, polished, and delved into through AI. Perhaps there is another possibility. That is, every conversation we have will become an exclusive learning assistant.

Well, alright, these are some of my current thoughts, and that's probably all. Obviously, if we continue to think, there will be many more ways to play. In short, I believe that everyone will have a large amount of session data to manage. The management of session data is a capability that current note taking tools do not possess. More importantly, as mentioned repeatedly earlier, ChatGPT next web is actually a decent note taking tool now.

Finally, I would like to attach my usage scenario:

![Snipaste_2023-11-27_20-30-30](https://github.com/Yidadaa/ChatGPT-Next-Web/assets/48642801/90a182cf-1b18-4a33-ac7d-5fb377dceeb8)
![Snipaste_2023-11-27_20-32-10](https://github.com/Yidadaa/ChatGPT-Next-Web/assets/48642801/7e8ade52-860b-470b-94c7-fb4c480d69c9)

As shown in the figure, I filled in the wrong API address in the settings to avoid calling AI. Then I input my feedback on ChatGPT from the dialog box. After receiving your reply, I added it. This completes a recording.

Oh, that's not right. I still used a note taking tool because my screenshots must be placed inside the note taking tool, After all, it's not possible to put pictures in ChatGPT next web yet (From this, it can be seen that when designing, if there is a note taking tool perspective, many functions may be designed differently, such as the image function. Who said that sending images must call AI? Can't we just save the images? Saving images is a note taking requirement, and the information on the images is useful to us, regardless of whether AI has the ability to interpret images.)

I think that if many details can be optimized and designed from the perspective of note taking tools, the final effect will definitely be very stunning. I am looking forward to it very much. Thank you!

**Comment by jkjoker** [comment]
> When users export content, timestamp information can also be attached.
> 
> This already added if you view a raw json

I'm sorry, many of my notes are in Chinese. I opened the exported JSON file and prepared to extract and edit some information from it. However, the Chinese text in it still couldn't be displayed correctly. I tried several editors, but none worked, such as vscode and notepad++. I'm not a technical professional. Can you give some advice? Thank you.

**Comment by H0llyW00dzZ** [comment]
> > When users export content, timestamp information can also be attached.
> > This already added if you view a raw json
> 
> I'm sorry, many of my notes are in Chinese. I opened the exported JSON file and prepared to extract and edit some information from it. However, the Chinese text in it still couldn't be displayed correctly. I tried several editors, but none worked, such as vscode and notepad++. I'm not a technical professional. Can you give some advice? Thank you.

if it still couldn't be displayed correctly. might related to this issue

[Bug] exported as a JSON file in the Windows installation version appears as unreadable content.（windows安装版的对话导出的JSON为乱码。） #3395

**Comment by Yidadaa** [comment]
1. 请使用专业的笔记工具，例如 Notion 或者 Obsidian，目前 next web 无法保证同步稳定性，仅适合作为聊天 UI；
2. 乱码问题 windows 编码格式有关，你应该使用文本编辑器的编码自动检测功能，从导出的 json 中识别正确的编码格式，你的乱码情况很有可能是因为编辑器尝试使用 gbk 格式读取了 utf-8 文件导致。
![image](https://github.com/Yidadaa/ChatGPT-Next-Web/assets/16968934/22e80870-1dff-42e6-8df5-ed44349f6c02)

**Comment by jkjoker** [comment]
> > Bot detected the issue body's language is not English, translate it automatically.
> 
> 1. Please use professional note-taking tools, such as Notion or Obsidian. Currently, next web cannot guarantee synchronization stability and is only suitable as a chat UI;
> 2. The garbled code problem is related to the Windows encoding format. You should use the automatic encoding detection function of the text editor to identify the correct encoding format from the exported json. Your garbled code situation is most likely because the editor is trying to read in gbk format. utf-8 file results.
>    ![image](https://private-user-images.githubusercontent.com/16968934/286470532-22e80870-1dff-42e6-8df5-ed44349f6c02.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEyNDM4OTUsIm5iZiI6MTcwMTI0MzU5NSwicGF0aCI6Ii8xNjk2ODkzNC8yODY0NzA1MzItMjJlODA4NzAtMWRmZi00MmU2LThkZjUtZWQ0NDM0OWY2YzAyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI5VDA3Mzk1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4YjU2NmRlYzY3MTVjMzkxMGU4MjdhZGNhM2JkZWRjZTM2NGE3NjM2ODM2MmE1MzVhYzg4NTdjM2RiMzZiY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._QXUkgXb6TAqZvfoSw31Rc4WtlN9Bs2V_jKseRG2Cg0)

谢谢，我再次尝试了下，关于格式问题，目前的情况是：
1、如果直接从客户端导出json到本地，那么，这个文件在vscode中打开时会提示“是二进制文件，或者编码格式不受支持”，强行打开会乱码；此外，我发现，从客户端本地导出的json文件，也无法导入回去。
2、如果是从同步到坚果云webdav的文件夹中获取json，则能正常打开，能正常显示，不存在编码问题。
